### ISSUE TITLE
bug: upload certs by apisix dashboard，the apisix pod will report error 

### ISSUE THREAD

**Issue by Cloud0916** [apache/apisix#11249]
### Current Behavior

config_etcd.lua:858: failed to fetch data from etcd: failed to check item data of [/apisix/ssls] err:additional properties forbidden, found validity_end,  etcd key: /apisix/ssls, context: ngx.timer 

### Steps to Reproduce

1.restart apisix 
2.delete certs and upload again

### Environment

all resources are deployed by helm
apisix: apache/apisix:3.9.1-debian
apisix-etcd: docker.io/bitnami/etcd:3.5.7-debian-11-r14
apisix-dashboard: apache/apisix-dashboard:3.0.0-alpine

**Comment by rahulracker7539** [comment]
am having the similar problem :- 
[error] 51#51: *1 [lua] config_etcd.lua:535: load_full_data(): failed to check item data of [/apisix/ssls] err:additional properties forbidden, found validity_start
@Cloud0916 did you find any solution ?

**Comment by Cloud0916** [comment]
> am having the similar problem :- [error] 51#51: *1 [lua] config_etcd.lua:535: load_full_data(): failed to check item data of [/apisix/ssls] err:additional properties forbidden, found validity_start @Cloud0916 did you find any solution ?

No,i just find that when i import cert file from dashboard, the json data stored in etcd contains "validity_start" and "validity_end".I guess is the version of dashboard and etcd not suitable，but i still not find any solution.

**Comment by emrocha** [comment]
In this Pull Request (https://github.com/apache/apisix/pull/10323) , they recognized that there was a break in compatibility with the Apisix Dashboard project.

There is a open issue about this https://github.com/apache/apisix-dashboard/issues/2915

**Comment by wangchao732** [comment]
I have a same.

**Comment by Cloud0916** [comment]
it's ok to upload cert by apisix-admin api, for example:
[root@server1 ~]# CERT=$(awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' your_domain.pem)
[root@server1 ~]# KEY=$(awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' your_domain.key)
curl http://{Adress of apisix-admin}:9180/apisix/admin/ssls -X POST -d '{
    "cert": "'"$CERT"'",
    "key": "'"$KEY"'",
    "snis": ["your.domain"]
}' -H 'X-API-KEY: {your api key}'

**Comment by wangchao732** [comment]
> 可以通过apisix-admin api上传证书，例如：[root@server1 ~]# CERT=$（awk 'NF {sub（/\r/， “”）;printf “%s\n”，$0;}'your_domain.pem） [root@server1 ~]# KEY=$(awk 'NF {sub(/\r/, ""); printf "%s\n",$0;}' your_domain.key) curl http://{Adress of apisix-admin}:9180/apisix/admin/ssls -X POST -d '{ "cert": "'"$CERT"'", "key": "'"$KEY"'", "snis": ["your.domain"] }' -H 'X-API-KEY: {your api key}'

yes,Through apisix /apisix/admin/ssls put certificate successfully, through the dashbord still does not work, apisix version 3.9.0, compared the results found that apisix dashbord upload field contains：

      {
			"createdIndex": 314,
			"key": "/apisix/ssls/514925433557353103",
			"modifiedIndex": 314,
			"value": {
				"validity_start": 1700092800,
				"id": "514925433557353103",
				"status": 1,
				"update_time": 1716449075,
				"snis": [
					"*.xxx.com",
					"xxx.com"
				],
				"cert": "",
				"create_time": 1716449075,
				"validity_end": 1731715199
			}
		}

but, api  hasn't "validity_start"

{
	"list": [
		{
			"createdIndex": 313,
			"key": "/apisix/ssls/00000000000000000312",
			"modifiedIndex": 313,
			"value": {
				"create_time": 1716448592,
				"update_time": 1716448592,
				"snis": [
					"*.bcnyyun.com"
				],
				"type": "server",
				"id": "00000000000000000312",
				"key": "",
				"status": 1
			}
		}
}

**Comment by fatihusta** [comment]
I added quick fix patch for apisix-dashboard. https://github.com/apache/apisix-dashboard/issues/2915#issuecomment-2127662316

**Comment by adangadang** [comment]
通过apisix-dashboard 页面导入证书(泛域名证书),会保找不到对应域名SNI, 将证书内容复制出来,通过 http://127.0.0.1:9180/apisix/admin/ssls/1 导入,可以正常使用

**Comment by fatihusta** [comment]
This issue fixed in apisix-dashboard master branch. 
https://github.com/apache/apisix-dashboard/pull/2947


Please go to build project and make apisix-dashboard package.
https://github.com/api7/apisix-build-tools?tab=readme-ov-file#build-dashboard


```
make package type=deb app=dashboard version=3.0.1 checkout=master
```

**Comment by fatihusta** [comment]
> > This issue fixed in apisix-dashboard master branch. [apache/apisix-dashboard#2947](https://github.com/apache/apisix-dashboard/pull/2947)
> > 
> > Please go to build project and make apisix-dashboard package. https://github.com/api7/apisix-build-tools?tab=readme-ov-file#build-dashboard
> > 
> > ```
> > make package type=deb app=dashboard version=3.0.1 checkout=master
> > ```
> > 
> > Please remove validity keys in payload If you are using apisix admin API.
> 
> Is the fix in the latest version of the `apisix-dashboard` Docker image? I was using `apache/apisix-dashboard:latest` and I was getting the validity error.


I guess not in latest version in hub. 
Please build your own image from this repo.

https://github.com/apache/apisix-docker.git

```sh
# change directory
cd apisix-docker/dashboard 
docker build \
  -f Dockerfile.alpine \
  --build-arg APISIX_DASHBOARD_TAG=master \
  -t apisix-dashboard:master \
  .

```

**Comment by 1450793561** [comment]
> ```
> make package type=deb app=dashboard version=3.0.1 checkout=master
> ```
I also got a similar error when I installed version 3.0.1 using yum. Has this RPM package not been fixed yet?
Do I have to recompile to generate rpm packages using make?

**Comment by fatihusta** [comment]
> > ```
> > make package type=deb app=dashboard version=3.0.1 checkout=master
> > ```
> I also got a similar error when I installed version 3.0.1 using yum. Has this RPM package not been fixed yet?
> Do I have to recompile to generate rpm packages using make?
> 

Yes you can compiled like deb. Just set type to rpm

**Comment by supeiliang** [comment]
> > > 该问题已在 apisix-dashboard 主分支中修复。apache [/apisix-dashboard#2947](https://github.com/apache/apisix-dashboard/pull/2947)
> > > 请前往构建项目并制作 apisix-dashboard 包。https [://github.com/api7/apisix-build-tools?tab =readme-ov-file#build-dashboard](https://github.com/api7/apisix-build-tools?tab=readme-ov-file#build-dashboard)
> > > ```
> > > make package type=deb app=dashboard version=3.0.1 checkout=master
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 如果您正在使用 apisix 管理 API，请删除有效负载中的有效性密钥。
> > 
> > 
> > 最新版本的 Docker 镜像中是否有修复`apisix-dashboard`？我正在使用`apache/apisix-dashboard:latest`，但遇到了有效性错误。
> 
> 我猜 Hub 的最新版本里没有。 请从这个仓库构建你自己的镜像。
> 
> https://github.com/apache/apisix-docker.git
> 
> # change directory
> cd apisix-docker/dashboard 
> docker build \
>   -f Dockerfile.alpine \
>   --build-arg APISIX_DASHBOARD_TAG=master \
>   -t apisix-dashboard:master \
>   .

使用这个dockerfile构建会有各种版本问题的报错，可以提供最新的dockerfile吗？

**Comment by Baoyuantop** [comment]
Older versions of apisix dashbaord may not work with the latest version of apisix, you can use the admin api to modify the configuration. We are building an apisix dashboard that works with the new version https://github.com/apache/apisix-dashboard/issues/2981. If there is still a problem, please open it again.
